components/datetime/src/provider/neo.rs

```diff
@@ -108,6 +110,29 @@ pub enum MonthSymbolsV1<'data> {
     ///
     /// Found for lunisolar and lunisidereal calendars
     LeapLinear(#[cfg_attr(feature = "serde", serde(borrow))] VarZeroVec<'data, str>),
+
+    /// This represents the formatting to apply to numeric values to produce the corresponding
+    /// leap month symbol.
+    ///
+    /// For numeric formatting only, on calendars with leap months
+    LeapNumeric(#[cfg_attr(feature = "serde", serde(borrow))] SimpleSubstitutionPattern<'data>),
+}
+
+/// Represents a simple substitution pattern;
+/// i.e. a string with a single placeholder
+#[derive(Debug, PartialEq, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(
+ feature = "datagen",
+ derive(serde::Serialize, databake::Bake),
+ databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct SimpleSubstitutionPattern<'data> {
+    /// The pattern
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub pattern: Cow<'data, str>,
+    /// The index in which to substitute stuff
```

Comment:
Nit: State that it is the byte index (not code point index, etc)

Nit: State that there is a weak invariant that the index is less than the byte length of the string